File: 0538-Convert BST to Greater Tree/0538-Convert BST to Greater Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* convertBST(TreeNode* root) {
        stack<TreeNode*> St;
        TreeNode* node = root;
        int presum = 0;
        while (node || !St.empty()) {
            if (node) {
                St.push(node);
                node = node->right;
            }
            else {
                node = St.top();
                St.pop();
                node->val += presum;
                presum = node->val;
                node = node->left;
            }
        }
        
        return root;
    }
};

File: 0501-0600/0538-Convert BST to Greater Tree/0538-Convert BST to Greater Tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def convertBST(self, root: TreeNode) -> TreeNode:
        St = []
        presum = 0
        node = root
        while St or node:
            if node:
                St.append(node)
                node = node.right
            else:
                node = St.pop()
                node.val += presum
                presum = node.val
                node = node.left
        return root
